internal/handlers/tigris/msg_update.go

```diff
@@ -195,24 +193,18 @@ func (h *Handler) MsgUpdate(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
 	return &reply, nil
 }
 
-// update updates documents by _id.
+// update replaces given document.
 func (h *Handler) update(ctx context.Context, sp fetchParam, doc *types.Document) (int, error) {
-	id := must.NotFail(doc.Get("_id"))
-	f := must.NotFail(filter.Eq("_id", id).Build())
-	h.L.Sugar().Debugf("Update filter: %s", f)
-
-	update := fields.UpdateBuilder()
-	for _, k := range doc.Keys() {
-		v := must.NotFail(doc.Get(k))
-		update.Set(k, json.RawMessage(must.NotFail(tjson.Marshal(v))))
+	u, err := tjson.Marshal(doc)
+	if err != nil {
+		return 0, lazyerrors.Error(err)
 	}
-	u := must.NotFail(update.Build()).Built()
 	h.L.Sugar().Debugf("Update: %s", u)
 
-	res, err := h.driver.UseDatabase(sp.db).Update(ctx, sp.collection, f, u)
+	_, err = h.driver.UseDatabase(sp.db).Replace(ctx, sp.collection, []driver.Document{u})
 	if err != nil {
 		return 0, lazyerrors.Error(err)
 	}
 
-	return int(res.ModifiedCount), nil
+	return 1, nil
```

Comment:
Isn't possible for Update to have more than 1 modified fields?

Reply:
We return modified docs count, not fields. As we always modify only 1 document we must return 1.

Reply:
Understandable, LGTM then!